src/api/ledger/pathfind.js

```diff
@@ -1,147 +1,99 @@
   }
+}
 
-  function formatPath(pathfindResults) {
-    const alternatives = pathfindResults.alternatives;
-    if (alternatives && alternatives.length > 0) {
-      return parsePathfind(pathfindResults);
-    }
-    if (pathfindResults.destination_currencies.indexOf(
-            pathfind.destination.amount.currency) === -1) {
-      throw new NotFoundError('No paths found. ' +
-        'The destination_account does not accept ' +
-        pathfind.destination.amount.currency +
-        ', they only accept: ' +
-        pathfindResults.destination_currencies.join(', '));
-    } else if (pathfindResults.source_currencies
-               && pathfindResults.source_currencies.length > 0) {
-      throw new NotFoundError('No paths found. Please ensure' +
-        ' that the source_account has sufficient funds to execute' +
-        ' the payment in one of the specified source_currencies. If it does' +
-        ' there may be insufficient liquidity in the network to execute' +
-        ' this payment right now');
-    } else {
-      throw new NotFoundError('No paths found.' +
-        ' Please ensure that the source_account has sufficient funds to' +
-        ' execute the payment. If it does there may be insufficient liquidity' +
-        ' in the network to execute this payment right now');
-    }
+function formatResponse(pathfind, paths) {
+  if (paths.alternatives && paths.alternatives.length > 0) {
+    const address = pathfind.source.address;
+    return parsePathfind(address, pathfind.destination.amount, paths);
   }
-
-  function formatResponse(payments) {
-    return {payments: payments};
+  if (!_.includes(paths.destination_currencies,
+      pathfind.destination.amount.currency)) {
+    throw new NotFoundError('No paths found. ' +
+      'The destination_account does not accept ' +
+      pathfind.destination.amount.currency + ', they only accept: ' +
+      paths.destination_currencies.join(', '));
+  } else if (paths.source_currencies && paths.source_currencies.length > 0) {
+    throw new NotFoundError('No paths found. Please ensure' +
+      ' that the source_account has sufficient funds to execute' +
+      ' the payment in one of the specified source_currencies. If it does' +
+      ' there may be insufficient liquidity in the network to execute' +
+      ' this payment right now');
+  } else {
+    throw new NotFoundError('No paths found.' +
+      ' Please ensure that the source_account has sufficient funds to' +
+      ' execute the payment. If it does there may be insufficient liquidity' +
+      ' in the network to execute this payment right now');
   }
+}
 
-  const steps = [
-    asyncify(prepareOptions),
-    findPath,
-    addDirectXrpPath,
-    asyncify(formatPath),
-    asyncify(formatResponse)
-  ];
+function getPathFind(pathfind, callback) {
+  validate.pathfind(pathfind);
 
-  async.waterfall(steps, callback);
+  const address = pathfind.source.address;
+  async.waterfall([
```

Comment:
Probably beyond the scope of this changeset, but wrt: direct XRP, it may be useful to check the account roots of the src and destination, for the (available after reserve) balance of the source and whether the destination accepts XRP. There's a flag that disallows it.